cypress/e2e/02a-bulk_upload_patient.cy.js

```diff
@@ -12,8 +17,27 @@ describe("Bulk upload patients", () => {
   loginHooks();
 
   before("setup data", () => {
-    let currentSpecRunVersionName = `${testNumber()}-cypress-${specRunName}`;
-    setupRunData(currentSpecRunVersionName);
+    cy.task("getSpecRunVersionName", specRunName)
+      .then((prevSpecRunVersionName) => {
+        if (prevSpecRunVersionName) {
+          cleanUpPreviousRunSetupData(prevSpecRunVersionName);
+          // putting this here as well as in the after hook to guarantee
+          // the cleanup function runs even if the test gets interrupted
+          cleanUpRunOktaOrgs(prevSpecRunVersionName);
+        }
+      })
+      .then(() => {
+        let data = {
+          specRunName: specRunName,
+          versionName: currentSpecRunVersionName,
+        };
+        cy.task("setSpecRunVersionName", data);
+        setupRunData(currentSpecRunVersionName);
+      });
+  });
+
+  after(() => {
+    cleanUpRunOktaOrgs(currentSpecRunVersionName);
```

Comment:
So I thought about this a bit and while I think the official Cypress suggestions to not use the after hook makes sense for internal state, I think we should still clean up Okta here since it's an external system / there's no guarantee that a re-run will be kicked for a subsequent run. I also stuck a cleanup in the before hook to at least partially handle the case where the test exits midway.